File: source/data/mappers/src/androidMain/kotlin/com/voxeldev/tgdrive/mappers/media/DocumentMapper.kt
package com.voxeldev.tgdrive.mappers.media

import org.drinkless.td.libcore.telegram.TdApi.Document
import com.voxeldev.tgdrive.main.api.media.Document as CommonDocument

/**
 * @author nvoxel
 */
class DocumentMapper(
    private val thumbnailMapper: ThumbnailMapper,
    private val fileMapper: FileMapper,
) {

    fun toResponse(document: Document): CommonDocument =
        CommonDocument(
            fileName = document.fileName,
            mimeType = document.mimeType,
            thumbnail = document.thumbnail?.let { thumbnailMapper.toResponse(thumbnail = document.thumbnail) },
            file = fileMapper.toResponse(file = document.document)
        )
}

File: iosApp/iosApp/data/mappers/media/DocumentMapper.swift
//
//  DocumentMapper.swift
//  iosApp
//
//  Created by nVoxel on 25.04.2024.
//  Copyright © 2024 orgName. All rights reserved.
//

import Foundation
import TDLibKit
import TGDriveKit

class DocumentMapper {
    
    private let thumbnailMapper: ThumbnailMapper
    private let fileMapper: FileMapper
    
    init(thumbnailMapper: ThumbnailMapper, fileMapper: FileMapper) {
        self.thumbnailMapper = thumbnailMapper
        self.fileMapper = fileMapper
    }
    
    func toResponse(document: Document) -> CommonDocument {
        return CommonDocument(
            fileName: document.fileName,
            mimeType: document.mimeType,
            thumbnail: document.thumbnail != nil ? thumbnailMapper.toResponse(thumbnail: document.thumbnail!) : nil,
            file: fileMapper.toResponse(file: document.document)
        )
    }
}
